code/parse/sexp.cpp

```diff
@@ -10249,14 +10258,34 @@ int sexp_num_valid_arguments( int n )
 
 	arg_handler = get_handler_for_x_of_operator(n);
 
+	// thanks to woutersmits for finding this loophole
```

Comment:
Which loophole!  My only issue with the PR is we don't know why they need special handling by just looking at the comments.

Reply:
Merely that using the SEXP outside of an -argument operator will cause a crash due to the node being -1.  I can make the comment more explicit.